packages/synapse-react-client/src/utils/jsonschema/SchemaAnnotationUtils.ts

```diff
@@ -86,6 +81,40 @@ export function getDisplayedAnnotationDescription(
   return undefined
 }
 
+/**
+ * Registers a schema with @hyperjump/json-schema.
+ * @param schema
+ * @return boolean indicating if the schema is registered.
+ */
+export function useRegisterSchema(schema?: JSONSchema7) {
+  const [isRegistered, setIsRegistered] = useState(false)
+
+  useEffect(() => {
+    if (schema && schema.$id) {
+      // The schema can be registered
+      if (!hasSchema(schema.$id)) {
+        try {
+          registerSchema(schema as SchemaObject)
+          setIsRegistered(true)
+        } catch (e) {
+          console.error(
+            'Unable to register JSON Schema in @hyperjump/json-schema:',
+            e,
+          )
+        }
```

Comment:
This `try`/`catch` handles the error causing the bug.

If this fails, we simply will not use the annotation enhancements (graceful degradation).